test/demo.jl

```diff
@@ -105,3 +105,58 @@ end
     @test "μ" ∈ labels
     @test "σ" ∈ labels
 end
+
+# -- test ADTypes compatibility
+
+@testset "Demo with ADTyes" begin
+    # Define the target distribution using the `LogDensityProblem` interface
```

Comment:
Can't we just reuse existing models? The main thing to test is just that you can use (different) ADTypes instead of Val to specify the backend. It shouldn't be necessary to design a custom model.